vendor/bat-native-ads/src/bat/ads/internal/ads/new_tab_page_ads/new_tab_page_ad.cc

```diff
@@ -100,14 +88,6 @@ void NewTabPageAd::FireEvent(const NewTabPageAdInfo& ad,
           return;
         }
 
-        if (event_type == mojom::NewTabPageAdEventType::kViewed) {
-          // TODO(https://github.com/brave/brave-browser/issues/14015): We need
-          // to fire an ad served event until new tab page ads are served by the
-          // ads library
-          FireEvent(uuid, creative_instance_id,
-                    mojom::NewTabPageAdEventType::kServed);
-        }
-
```

Comment:
This logic is still required if ads are disabled

Reply:
added condition to check if brave ads are disabled:
https://github.com/brave/brave-core/pull/11945/files#diff-a0bccac6c1f49e72263568f7d661a470ea0673cb351f4060fb817b57a446407fR106